### PR TITLE
Return error instead of logging it

### PR DESCRIPTION
**Pull request by lookuptable**
This allows caller to deal with errors directly and makes testing
easier.

### CODE REVIEW

File: pkg/pki/crypto_test.go [hunk 1]
```diff
@@ -60,76 +88,99 @@ iYh1iBeqIBCSLPt8SrpAWGyOMKLN4bMCWFNOaeBFLG/91K8Yo0swSTAOBgNVHQ8B
...
+	for id, c := range testCases {
+		key, err := ParsePemEncodedKey(c.algo, []byte(c.pem))
+		if c.errMsg != "" {
+			if err == nil {
+				t.Errorf("%s: no error is returned", id)
+			} else if c.errMsg != err.Error() {
+				t.Errorf(`%s: Unexpected error message: want "%s" but got "%s"`, id, c.errMsg, err.Error())
+			}
+		} else {
+			if keyType := reflect.TypeOf(key); keyType != c.keyType {
```
Comment (wattli) [inline pkg/pki/crypto_test.go]: combine this 
else {
   if 

to "else if" directly?

File: pkg/pki/crypto.go [hunk 2]
```diff
@@ -18,48 +18,47 @@ import (
...
+	"fmt"
 )
 
 // ParsePemEncodedCertificate constructs a `x509.Certificate` object using the
 // given a PEM-encoded certificate,
-func ParsePemEncodedCertificate(certBytes []byte) *x509.Certificate {
+func ParsePemEncodedCertificate(certBytes []byte) (*x509.Certificate, error) {
 	cb, _ := pem.Decode(certBytes)
 	if cb == nil {
-		glog.Fatalf("Invalid PEM encoding for the certificate: %s", certBytes)
```
Comment (wattli) [inline pkg/pki/crypto.go]: Good work, i remember Fatalf is mostly used for main file

Comment (lookuptable) [inline pkg/pki/crypto.go]: yeah that's the objective for this PR.

File: pkg/pki/crypto_test.go (hunk 1 above)
Comment (lookuptable) [inline pkg/pki/crypto_test.go]: Done

**Pull request merged**